src/hotspot/share/runtime/sharedRuntime.cpp

```diff
@@ -1597,38 +1601,15 @@ JRT_END
 // resolve a static call and patch code
 JRT_BLOCK_ENTRY(address, SharedRuntime::resolve_static_call_C(JavaThread* current ))
   methodHandle callee_method;
-  bool enter_special = false;
   JRT_BLOCK
     callee_method = SharedRuntime::resolve_helper(false, false, CHECK_NULL);
     current->set_vm_result_2(callee_method());
-
-    if (current->is_interp_only_mode()) {
-      RegisterMap reg_map(current,
-                          RegisterMap::UpdateMap::skip,
-                          RegisterMap::ProcessFrames::include,
-                          RegisterMap::WalkContinuation::skip);
-      frame stub_frame = current->last_frame();
-      assert(stub_frame.is_runtime_frame(), "must be a runtimeStub");
-      frame caller = stub_frame.sender(&reg_map);
-      enter_special = caller.cb() != nullptr && caller.cb()->is_compiled()
-        && caller.cb()->as_compiled_method()->method()->is_continuation_enter_intrinsic();
-    }
   JRT_BLOCK_END
-
-  if (current->is_interp_only_mode() && enter_special) {
-    // enterSpecial is compiled and calls this method to resolve the call to Continuation::enter
-    // but in interp_only_mode we need to go to the interpreted entry
-    // The c2i won't patch in this mode -- see fixup_callers_callsite
-    //
-    // This should probably be done in all cases, not just enterSpecial (see JDK-8218403),
-    // but that's part of a larger fix, and the situation is worse for enterSpecial, as it has no
-    // interpreted version.
-    return callee_method->get_c2i_entry();
-  }
```

Comment:
If I understand the analysis correctly, all this logic that we have to clearly
identify the calling context as ` jdk.internal.vm.Continuation.enterSpecial`
is no longer needed because we're switching to a more general fix of:
```
    if (current->is_interp_only_mode()) {
      target = callee_method->get_c2i_entry();
    } else {
      target = callee_method->verified_code_entry();
    }
    return target;
```
And that more general fix cover the special case code that we're deleting
(pun intended).

Reply:
Exactly. This code was added to fix a related issue, where a JavaThread already noticed it was in interpreted only mode but because enterSpecial doesn't have an actual interpreted version we switched from interpreted to compiled and from there we incorrectly called the compiled version of enter() leading to asserts in JVMTI code about stack depth. This fix is more general so it already covers that case too.